syntax = "proto3";

service Auth {
	rpc Login(LoginReq) returns (UserSession) {}
}

message LoginReq {
	optional string username = 1;
}

message UserSession {
	optional string session_token = 1;
}

service Meta {
	rpc Login2(Empty) returns (UserData) {}
	rpc GetUserData(Empty) returns (UserData) {}
	rpc GetShopItems(Empty) returns (ItemsList) {}
	rpc BuyItem(BuyItemReq) returns (Empty) {}
	rpc SellItem(SellItemReq) returns (Empty) {}
}

message Empty {}

message Item {
	optional string name = 1;
	optional int64 price = 2;
}

message GetUserInventoryReq {}
message ItemsListReq {}
message BuyItemReq {}
message SellItemReq {}

message UserData {
	optional string username = 1;
	optional int32 credits = 2;
	repeated Item items = 3;
}

message ItemsList {
	repeated Item items = 1;
}